src/coord/src/catalog/builtin.rs

```diff
@@ -1593,7 +1595,14 @@ pub const PG_INDEX: BuiltinView = BuiltinView {
     true::pg_catalog.bool AS indisvalid,
     -- MZ doesn't support replication so indisreplident is filled with false
     false::pg_catalog.bool AS indisreplident,
-    pg_catalog.array_agg(mz_index_columns.on_position ORDER BY mz_index_columns.index_position) AS indkey
+    -- Return zero if the index attribute is not a simple column reference, column position otherwise
+    pg_catalog.string_agg(coalesce(mz_index_columns.on_position, 0)::pg_catalog.text, ' ' ORDER BY mz_index_columns.index_position)::pg_catalog.int2vector AS indkey,
+    -- MZ doesn't have per-column flags, so returning a 0 for each column in the index
+    pg_catalog.string_agg('0', ' ')::pg_catalog.int2vector AS indoption,
+    -- Index expressions are returned in MZ format
+    pg_catalog.string_agg(mz_index_columns.on_expression, ' ' ORDER BY mz_index_columns.index_position) AS indexprs,
```

Comment:
Can `on_expression` ever contain a space or some vector delimiter? Column names can contain spaces so maybe? Might have to escape this.

Reply:
It can, e.g. `i::[s1004 AS pg_catalog.text]`. What would be the best way to escape this in SQL?
That said, this is a `pg_node_tree` in Postgres, so I'm not sure how machine readable we expect this to be, given that the Postgres equivalent becomes this:
```
({RELABELTYPE :arg {VAR :varno 1 :varattno 2 :vartype 1043 :vartypmod 34 :varcollid 100 :varlevelsup 0 :varnosyn 1 :varattnosyn 2 :location 38} :resulttype 25 :resulttypmod -1 :resultcollid 100 :relabelformat 1 :location 33})
```

Reply:
I don't have a good mental model of `pg_node_tree` besides cockroach and us both decided to just cast exprs to strings and shrug. It seems though that each expr is just wrapped in `{}` which is sufficient to disambiguate. We should perhaps do *something* here, even if it's not the fully correct way, just to get this PR in. I'm fine with one of: 1) manually wrap things in `{}` somehow, 2) figure out how properly escape these exprs into strings (I'm not aware of a way in SQL to do that?), 3) something else.

Reply:
What do you think [of it](https://github.com/MaterializeInc/materialize/pull/11075/files#diff-783c3cb9eeb2672c21407d2660db934c91386d13635b7281303a6446864427c2R63) now?
```
{t::[s1039 AS pg_catalog.varchar]}, {i::[s1004 AS pg_catalog.text]}
```